enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,297 @@
+MicroShift deployment (the list of active plugins can be obtained by running
+`sos report -l`). However, there are some specifics that we need to add to a
+new plugin:
+1. Control plane component logs: apiserver, etcd, kubelet, etc. (these are
+retrieved by syslog, meaning they are mixed with other system logs).
+1. Yaml output for relevant manifests from API server.
+1. Logs for every container. These are already retrieved by CRIO plugin. May
+need easier indexing.
+1. MicroShift's configuration and version. This [enhancement](https://github.com/openshift/enhancements/pull/1304)
+might add additional information to gather: plugins, manifests, etc.
+
+Retrieving all this information involves using a kubernetes client. While
+MicroShift itself uses the go client extensively, the idea of introducing
+another command to produce this output and avoid installing `oc`/`kubectl` is
+not the best, since we would be adding runtime dependencies between a system
+component and MicroShift.
+
+### Workflow Description
+
+> _Disclaimer:_ Workflows described here illustrate manual processes.
+
+Assuming from now on MicroShift has already been installed in a R4E/RHEL based
+system and there has been some kind of issue.
+
+1. MicroShift administrator notices an issue/wants to get a system report.
+1. The administrator will log in to the failing host.
+1. The administrator will perform the debug report creation procedure:
+```bash
+sudo sos report
+```
+
+For the CI, the same procedure applies.
+
+#### Variation [optional]
+N/A
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+### Risks and Mitigations
+There are four main risks:
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. Extra required disk space.
+1. CEE having a different tool.
+
+#### Upgrades
+When upgrading a MicroShift system we may need to upgrade just MicroShift, or
+the OS as well. If it is the latter then `sos` will be updated before
+MicroShift, therefore we must ensure backwards compatibility with the plugin.
+
+If it is the former then we need to ensure newer MicroShift versions do not
+introduce incompatible changes with the reporting capabilities of the `sos`
+plugin. This may also be seen from a different perspective: the plugin should
+make no assumptions about names or locations of components whenever possible.
+In other words, the plugin must be written in the most generic way that still
+allows retrieval of relevant information.
+
+Still, if incompatible changes are introduced then we need to upgrade `sos` as
+well.
+
+#### Sos release cadence
+`sos` release cadence has been shortened to 4 weeks starting on Feb 1st 2023.
+This change has been introduced to allow faster support of new components and
+features. Since MicroShift follows the same release calendar as OpenShift,
+`sos` is faster.
+
+New versions of `sos` will simply slot in to the next regular release. If there
```

Comment:
The detail that was important to me was that releases go into both 8 & 9.

```suggestion
New versions of `sos` will simply slot in to the next regular release of both RHEL 8 and 9. If there
```

Reply:
Originally included this in the version skew section. Adding it here too.